app/Jobs/Bank/ProcessBankTransactionsNordigen.php

```diff
@@ -0,0 +1,183 @@
+ *
+ * @copyright Copyright (c) 2022. Credit Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\Jobs\Bank;
+
+use App\Helpers\Bank\Nordigen\Nordigen;
+use App\Libraries\MultiDB;
+use App\Models\Account;
+use App\Models\BankIntegration;
+use App\Models\BankTransaction;
+use App\Models\Company;
+use App\Notifications\Ninja\GenericNinjaAdminNotification;
+use App\Services\Bank\BankMatchingService;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Log;
+
+class ProcessBankTransactionsNordigen implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    private BankIntegration $bank_integration;
+
+    private ?string $from_date;
+
+    public Company $company;
+    public Nordigen $nordigen;
+    public $nordigen_account;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(BankIntegration $bank_integration)
+    {
+        $this->bank_integration = $bank_integration;
+        $this->from_date = $bank_integration->from_date ?: now()->subDays(90);
+        $this->company = $this->bank_integration->company;
+    }
+
+    /**
+     * Execute the job.
+     *
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        if ($this->bank_integration->integration_type != BankIntegration::INTEGRATION_TYPE_NORDIGEN)
+            throw new \Exception("Invalid BankIntegration Type");
+
+        if (!(config('ninja.nordigen.secret_id') && config('ninja.nordigen.secret_key')))
+            throw new \Exception("Missing credentials for bank_integration service nordigen");
+
+        $this->nordigen = new Nordigen();
+
+        set_time_limit(0);
+
+        nlog("Nordigen: Processing transactions for account: {$this->bank_integration->account->key}");
+
+        // UPDATE ACCOUNT
+        try {
+            $this->updateAccount();
+        } catch (\Exception $e) {
+            nlog("Nordigen: {$this->bank_integration->nordigen_account_id} - exited abnormally => " . $e->getMessage());
+
+            $content = [
+                "Processing transactions for account: {$this->bank_integration->nordigen_account_id} failed",
+                "Exception Details => ",
+                $e->getMessage(),
+            ];
+
+            $this->bank_integration->company->notification(new GenericNinjaAdminNotification($content))->ninja();
+
+            throw $e;
+        }
+        if (!$this->nordigen_account)
+            return;
+
+        // UPDATE TRANSACTIONS
+        try {
+            $this->processTransactions();
+        } catch (\Exception $e) {
+            // reset from_date in case this was the error (self-heal) and perform a max sync of data we can fetch (next-time) @todo we should analyze the error for this
```

Comment:
I think for Nordigen the most common scenario may be a revoked requisition ? 

There should be checks prior to arriving in this method for that particular scenario (disabled_upstream). So i would assume any failures at this point may cause the disabled_upstream flag to be foggled.

Reply:
the eypired requisition is checked by the isAccountActive Section and we would never reach the Transactions section.

The user can manualy set the fram_date in the portal, so its possible he/she sets it to 3 years ago

Reply:
i reproduced exactly this behavior and nordigen does not return any errors and silently just provides the data within the available time-range.
Should we leave the reset there or remove it?

Reply:
If it doesn't do anything useful, we can remove it.

Reply:
fixxed